BipedRobot/ZMPBipedRobot/src/WalkingOptimization.jl

```diff
@@ -0,0 +1,382 @@
+"""
+
+A Walking Optimization perform solve an optimisation problem based on : 
+    `x = argmin/argmax f(x)`
+    `g(x) <= 0`
+    `h(x) = 0`
+
+where `g(x)` and `h(x)` are constraints functions and `x` is an candidate (ZMP based controller)
+The methodology used here is to simulate each candidate from a discrete input-space. 
+
+A `WalkingOptimization` has 2 differents outputs :
+    1.  `x` which is the value of a candidate 
+    2.  `f` which is the objective value of the candidate `x`
+
+"""
+mutable struct WalkingOptimization
+    # Input space
+    Δz::Real
+    Tstep::Real
+    Lmax::Real
+    δ::Real
+    Tver::Real
+    hstep::Real
+
+    # Auto defined input space
+    zc::Real
+    previewTime::Real
+
+    # Output space 
+    energy::Real
+    walkingSpeed::Real
+end
+
+"""
+
+Compute the energy from a torque and joint speed vector 
+"""
+function computeEnergy(torque::Vector{T}, omega::Vector{T}, dt::T) where {T <: Real}
+    energy = zero(T)
+    n = length(torque)
+    power = zeros(T, n)
+    torque = abs.(torque)
+    omega = abs.(omega)
+    for i in 1:(n - 1)
+        avg_torque = (torque[i] + torque[i + 1]) / 2
+        avg_omega = (omega[i] + omega[i + 1]) / 2
+        energy += avg_torque * avg_omega * dt
+        power[i] = avg_torque * avg_omega
+    end
+    return energy, power
+end
+
+"""
+
+Return a `BipedRobot` data structure based on a `WalkingOptimization` data structure 
+"""
+function defineBipedRobot(wo::WalkingOptimization)
+    include(joinpath(packagepath(), "param.jl"))
+    urdfpath() = joinpath(packagepath(), "ZMP_2DBipedRobot.urdf")
+    Δz = wo.Δz
+    Tstep = wo.Tstep
+    Lmax = wo.Lmax
+    δ = wo.δ
+    Tver = wo.Tver
+    hstep = wo.hstep
+    zc = wo.zc
+    previewTime = wo.previewTime
+    ΔCoMz = 0.21902787835056647 - zc
```

Comment:
Isn't this already in the `param.jl` files? Let's try to avoid all code duplication